spec/features/staff/users_can_view_an_activity_as_xml_spec.rb

```diff
@@ -42,6 +42,27 @@
         end
       end
 
+      context "when the activity does not have actual dates (optional dates)" do
+        let(:activity) {
+          create(:fund_activity,
+            organisation: organisation,
+            identifier: "IND-ENT-IFIER",
+            actual_start_date: nil,
+            actual_end_date: nil)
+        }
+        let(:activity_presenter) { ActivityXmlPresenter.new(activity) }
+        let(:xml) { Nokogiri::XML::Document.parse(page.body) }
+
+        it "does not include empty optional dates" do
+          visit organisation_activity_path(organisation, activity, format: :xml)
+          optional_start_date = xml.at("iati-activity/activity-date[@type = '2']")
+          optional_end_date = xml.at("iati-activity/activity-date[@type = '4']")
+
+          expect(optional_start_date.present?).to eq(false)
```

Comment:
Given your `xml.at` returns nil to pass, what do you think about loosing the `present?` and being explicit? Something like:

```` ruby
expect(optional_start_date).to be_nil
````

Reply:
That's a really good point! This was my first full-on TDD experience on my own and I struggled to create the test. I followed other examples and then I added what I needed, but definitely I could've done my life easier as you suggest. I didn't consider that nil will be the same than "not present". Happy to make this change as I think it's easier to read

Reply:
Well, I think you are right `nil.present?`does equal false, I just wonder if it is needed. 

This xml stuff is hard - you've done a great job on your own! 👍